### ISSUE TITLE
Images don't show up when attached to email

### ISSUE THREAD

**Issue by blakkheimgw** [RainLoop/rainloop-webmail#1253]
**RainLoop version, browser, OS:**
1.10.5, Chrome/Firefox, Linux/Windows

**Expected behavior and actual behavior:**
Images should be displayed in email. It doesn't and images are replaced by some grey space.

**Steps to reproduce the problem:**
Try to display an email with images attached to it (images linked from a website will show up)

**Logs or screenshots:**
Grey space instead of image :
![selection_019](https://cloud.githubusercontent.com/assets/5330629/20094111/2011ae60-a5a0-11e6-9a85-c4d5468679d2.jpg)

**Comment by ervee** [comment]
Don't have this problem. Using FF ESR 45.4.0 on Windows

![capture](https://cloud.githubusercontent.com/assets/4435930/20094537/00df3fba-a5a2-11e6-9c61-6a2142bf625f.JPG)

Shows up in email I send to myself. During composing, in send items and in inbox.

**Comment by blakkheimgw** [comment]
Hmmm, weird. All sent messages (saved in Sent folder) or received messages don't show images. I forgot to mention that the problem appears after a Rainloop update : 1.10.3 to 1.10.4

Another thing : when I save a message in drafts and if I reopen it, my signature image is greyed out too.

**Comment by ervee** [comment]
Try updating to the latest: 1.10.5.192.

**Comment by blakkheimgw** [comment]
I'm already up to date : RainLoop 1.10.5.192

**Comment by ervee** [comment]
Then I'm out of ideas right now. Perhaps @RainLoop can quote on this.

**Comment by megadr01d** [comment]
Adblocker?

**Comment by blakkheimgw** [comment]
I tried with and without it with no noticeable effect.

**Comment by megadr01d** [comment]
Try right-clicking on the image, call the Web Inspector and see if the HTML wasn't changed by some browser plugin or something. Also try opening the image in a new tab/window (get the URL from right click or Web Inspector).

First, we need to see if it's an issue with Rainloop or your browser.

Also, running any Rainloop plugins? Try enabling/disabling the [local proxy for external images](https://github.com/RainLoop/rainloop-webmail/issues/1153) option, somewhere in Rainloop (admin area maybe, can't recall).

**Comment by blakkheimgw** [comment]
Hello,

All these images are base64 encoded. When inspecting the form of URL I noticed that the base64 payload is truncated and way too short. For example : data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsQAAA7EAZUrDhsAAAANSURBVBhXYzh8+PB/AAffA0nNPuCLAAAAAElFTkSuQmCC

**Comment by megadr01d** [comment]
Food for thought...

https://sendgrid.com/blog/embedding-images-emails-facts/
https://www.campaignmonitor.com/blog/email-marketing/2013/02/embedded-images-in-html-email/

> Inline Embedding (Base64 Encoding)
>
> Pros
> - Much simpler to achieve
> - Much faster to do
> - Requires much less deep dive into MIME and application code
>
> Cons
> - Can really increase size of emails especially if you use more than one image
> - Most likely blocked by default in many webmail services
> - Blocked completely in Outlook

**Comment by megadr01d** [comment]
Upon further testing, I think this is related to the browser. For example, the first image (You Tube) on [this](http://nattyornot.com/the-death-of-the-natural-bodybuilder-on-youtube/) post (which I get in my Inbox as part of my RSS feeds) always shows up as a gray placeholder. Sometimes, scrolling down the message shows other images as gray but as I keep scrolling they load up. Other times I have to refresh the browser for them to load.

This happens in Chrome and Firefox, Windows.
At home, on Safari, OS X, all images show up correctly.

**Comment by blakkheimgw** [comment]
I never see these blank images outside of my Rainloop webmail. Nor on Chromium, nor on Firefox.

I already encountered this problem before, after a Rainloop upgrade (fixed in 1.9.4 version if I remember well). And this time, this is the same scenario : after an upgrade (1.10.5), the problem appeared.

**Comment by megadr01d** [comment]
I explained it wrong. It's related to Rainloop and the browser. I never get this blank images outside of Rainloop either. But on Safari, I always get all images, inside or outside Rainloop.

And now that you mention it, I remember always having this problem of some images appearing gray on scolling (specially long messages) but scrolling up and down or refreshing the browser eventually loaded them up, so I just thought it to be a "natural flaw" in Rainloop fetching remote images.

**Comment by ervee** [comment]
I now also face this issue when sending mail with inline images pasted from the clipboard. Sometimes it shows no image in my sent item / receiver inbox and sometimes the gray box where the image should be. Difference in source code of the e-mail is one or two base64 encoded parts for the image. Why two? Don't know but they are different and that e-mail shows the gray box, the other with one part for the image shows nothing.

@RainLoop, do you want the .eml for the two emails?

**Comment by ervee** [comment]
This is actually a duplicate of #1252. @blakkheimgw, can you close this one?

**Comment by dtiUFVJM** [comment]
In my tests the problem occurs with png images larger than 60KB when inserted into the body of the message.
